[studio-ui] Updating LegacyFormDialog;

aachavarria opened the pull request:
Updating LegacyFormDialog.tsx to show it on a Dialog;
Adding success messages on wcm_content.js and wcm-dashboardwidget-common.js Edit;

aachavarria committed:
Adding saved as draft message;
Updating LegacyFormDialog.tsx to show it on a Dialog;
Adding success messages on wcm_content.js and wcm-dashboardwidget-common.js Edit;

rart commented on ui/app/src/state/epics/system.ts:
```diff
@@ -83,11 +83,17 @@ const systemEpics: CrafterCMSEpic[] = [
+      tap(({ payload: { action } }) => {
+        let message;
+        if (['save', 'saveAndMinimize'].includes(action)) {
+          message = getIntl().formatMessage(itemSuccessMessages.itemSavedAsDraft);
```
These two actions do regular save. Why showing save draft message?

aachavarria replied:
fixed

rart commented on ui/app/src/utils/i18n-legacy.ts:
```diff
@@ -1439,6 +1439,10 @@ export const itemSuccessMessages = defineMessages({
+  },
+  itemSavedAsDraft: {
+    id: 'item.savedAsDraft',
+    defaultMessage: 'Item saved as draft'
```
Item draft saved

aachavarria committed:
Deleting old Dialogs;

aachavarria committed:
Deleting old code, it belongs to old copy/paste api;

aachavarria committed:
Merge branch 'bugfix/LegacyPathNavigatorUpdates' of https://github.com/aachavarria/studio-ui into feature/legacyEditForms

aachavarria committed:
Removing SaveAndMinimize;
Adding savedAsDraft to systemPropsList;
Adding RELOAD_REQUEST to LegacyConcierge.tsx;
Updating LegacyCodeEditorDialog.tsx;

aachavarria committed:
updating wcm_content.ts to use LegacyEditForm

aachavarria committed:
adding LegacyEditForm Dialog to wcm_content.js and wcm-dashboardwidget-common.js;

aachavarria committed:
typo

aachavarria committed:
EditFormLegacy adding validation when form is minimized

rart commented on ui/app/src/utils/i18n-legacy.ts:
```diff
@@ -690,17 +686,17 @@ export const browseSearchMessages = defineMessages({
 export const formEngineMessages = defineMessages({
+  inProgressConfirmation: {
+    id: 'formEngine.inProgressConfirmation',
+    defaultMessage: 'There are an edit in progress, to open a new edit form please close or save the previous one'
```
Message: "Another form is currently open & minimized. Please close the current form before opening another."

We need this case for template editor too.

rart commented on ui/app/src/utils/codebase-bridge.tsx:
```diff
@@ -245,7 +245,11 @@ export function createCodebaseBridge() {
+              mountGlobalDialogManager={!isLegacy}
+              mountSnackbarProvider={!isLegacy}
+              mountLegacyConcierge={isLegacy}
+            >
```
PR mixup 😣

aachavarria replied:
this is already on develop, should not be listed here

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/legacyEditForms

rart commented on static-assets/components/cstudio-forms/forms-engine.js:
```diff
@@ -771,7 +772,7 @@ var CStudioForms =
 
     function saveDraftDialog() {
-      $.notify(formatMessage(formEngineMessages.saveDraftCompleted), 'success');
+      // deprecated
```
?

aachavarria committed:
adding codeEditor Minimized validation;

aachavarria committed:
removing old code saveDraftDialog();

rart commented on ui/app/src/utils/i18n-legacy.ts:
```diff
@@ -534,6 +534,10 @@ export const siteComponentDSMessages = defineMessages({
 export const codeEditorMessages = defineMessages({
+  inProgressConfirmation: {
+    id: 'codeEditor.inProgressConfirmation',
+    defaultMessage: 'Another form is currently open & minimized. Please close the current form before opening another'
```
"Another **editor** is currently..."

rart commented on ui/app/src/state/epics/dialogs.ts:
```diff
@@ -125,6 +130,58 @@ const dialogEpics: CrafterCMSEpic[] = [
+          return of(
+            showConfirmDialog({
+              body: getIntl().formatMessage(formEngineMessages.inProgressConfirmation)
+            })
```
The confirm dialog presents a question and this isn't one. We need to change the graphic and only the accept button should show up. See Confirm Dialog designs on Figma (3.2 project) for guidance.

We need to tweak ConfirmDialog to allow graphic swapping.
```jsx
function ConfirmDialogWrapper(props: ConfirmDialogProps) {
  const { onOk, onCancel, body, title, children, classes, imageUrl = questionGraphicUrl } = props;
  const { formatMessage } = useIntl();
  useUnmount(props.onClosed);
  return (
    <>
      <DialogContent id="confirmDialogBody" className={classes.dialogBody}>
        <img src={imageUrl} alt="" />
```

aachavarria committed:
adding imageUrl to ConfirmDialog.tsx;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/legacyEditForms

rart commented on ui/app/src/utils/i18n-legacy.ts:
```diff
@@ -534,6 +534,10 @@ export const siteComponentDSMessages = defineMessages({
 export const codeEditorMessages = defineMessages({
+  inProgressConfirmation: {
+    id: 'codeEditor.inProgressConfirmation',
+    defaultMessage: 'Another editor is currently open & minimized. Please close the current form before opening another'
```
"Please close the current **editor** before opening another"

rart commented on ui/app/src/state/epics/dialogs.ts:
```diff
@@ -125,6 +131,59 @@ const dialogEpics: CrafterCMSEpic[] = [
+        } else {
+          return of(
+            showConfirmDialog({
+              body: getIntl().formatMessage(codeEditorMessages.inProgressConfirmation)
```
info graphic

rart replied:
Why is this resolved? It isn’t fixed yet.

aachavarria committed:
renaming

rart commented on ui/app/src/utils/i18n-legacy.ts:
```diff
@@ -690,17 +691,18 @@ export const browseSearchMessages = defineMessages({
+  inProgressConfirmation: {
+    id: 'formEngine.inProgressConfirmation',
+    defaultMessage:
+      'Another editor is currently open & minimized. Please close the current editor before opening another'
```
This is a form. The other is an editor. This was was fine saying form in both sentences. The one above should say editor in both sentences.

aachavarria committed:
adding imageUrl to showConfirmDialog on Show Editor Dialog;
typo

sumerjabri commented:
@aachavarria please fix conflicts

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/legacyEditForms

 Conflicts:
	ui/app/src/components/SystemStatus/GlobalDialogManager.tsx

aachavarria commented:
> @aachavarria please fix conflicts

done
